src/NHSD.BuyingCatalogue.Identity.Api/Services/ProfileService.cs

```diff
@@ -61,26 +62,29 @@ public async Task GetProfileDataAsync(ProfileDataRequestContext context)
             if (user is not null)
             {
                 var claims = GetClaimsFromUser(user);
-                var primaryOrganisationName = await GetPrimaryOrganisationName(user.PrimaryOrganisationId);
-                if (primaryOrganisationName is not null)
-                {
-                    claims.Add(primaryOrganisationName);
-                }
+                claims.AddRange(await GetPrimaryOrganisationClaims(user.PrimaryOrganisationId));
 
                 context.IssuedClaims = claims;
             }
         }
 
-        public async Task<Claim> GetPrimaryOrganisationName(Guid primaryOrganisationId)
+        public async Task<IEnumerable<Claim>> GetPrimaryOrganisationClaims(Guid primaryOrganisationId)
         {
-            Claim claim = null;
-            var organisationName = (await organisationRepository.GetByIdAsync(primaryOrganisationId))?.Name;
-            if (!organisationName.IsNullOrEmpty())
+            List<Claim> claims = new List<Claim>();
+
+            var primaryOrganisation = await organisationRepository.GetByIdAsync(primaryOrganisationId);
+
+            if (primaryOrganisation is not null)
             {
```

Comment:
I think we should invert this `if` – I think the method will be slightly more readable.

Reply:
inverting the if gives you something like this
```c#
        public async Task<IEnumerable<Claim>> GetPrimaryOrganisationClaims(Guid primaryOrganisationId)
        {
            List<Claim> claims = new List<Claim>();

            var primaryOrganisation = await organisationRepository.GetByIdAsync(primaryOrganisationId);

            if (primaryOrganisation is null)
            {
                return claims;
            }

            if (!primaryOrganisation.Name.IsNullOrEmpty())
            {
                claims.Add(new Claim(ApplicationClaimTypes.PrimaryOrganisationName, primaryOrganisation.Name));
            }

            claims.AddRange(primaryOrganisation.RelatedOrganisations.Select(ro => new Claim(ApplicationClaimTypes.RelatedOrganisationId, ro.OrganisationId.ToString())));

            return claims;
        }
```
which imo doesn't look as good and adds extra lines of code

Reply:
I prefer that, although I don't slap braces around single line `if` statements either 😃, so I'd personally format it like this:

```c#
public async Task<IEnumerable<Claim>> GetPrimaryOrganisationClaims(Guid primaryOrganisationId)
{
    List<Claim> claims = new List<Claim>();

    var primaryOrganisation = await organisationRepository.GetByIdAsync(primaryOrganisationId);

    if (primaryOrganisation is null)
        return claims;

    if (!primaryOrganisation.Name.IsNullOrEmpty())
        claims.Add(new Claim(ApplicationClaimTypes.PrimaryOrganisationName, primaryOrganisation.Name));

    claims.AddRange(primaryOrganisation.RelatedOrganisations.Select(
        ro => new Claim(ApplicationClaimTypes.RelatedOrganisationId, ro.OrganisationId.ToString())));

    return claims;
}
```
However, I know some people have irrational fears about brace-less `if` statements! Ultimately though I'm not that fussed – leave it as is if you prefer.

Reply:
its not the brace-less if statement, i have a personal preference for having a single return path (if possible ofc) so you always know that it's coming out of one place. but your call tbh :)

Reply:
Ah, the old single exit point approach – that's old school. I'm not convinced it's really relevant these days – at least not in a well-written method.

Let's go with the inverted `if` – not because I want to wind you up but because I think it's probably more in-keeping with style used elsewhere. 😃